File: objective_c/UNOogiriTest/UNOogiriTest/ViewController.m
//
//  ViewController.m
//  UNOogiriTest
//
//  Created by B02681 on 2014/09/30.
//  Copyright (c) 2014年 B02681. All rights reserved.
//

#import "ViewController.h"


@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    // タブバーを生成
    self.tabView = [[UNTabScrollView alloc] init];
    
    NSArray *buttonList = @[@{@"title" : @"1", @"color" : @"1"},
                            @{@"title" : @"2", @"color" : @"1"},
                            @{@"title" : @"3", @"color" : @"1"},
                            @{@"title" : @"4", @"color" : @"1"},
                            @{@"title" : @"5", @"color" : @"1"},
                            @{@"title" : @"6", @"color" : @"1"},
                            @{@"title" : @"7", @"color" : @"1"},
                            @{@"title" : @"8", @"color" : @"1"},
                            @{@"title" : @"9", @"color" : @"1"},
                            @{@"title" : @"10", @"color" : @"1"},
                            @{@"title" : @"11", @"color" : @"1"},
                            @{@"title" : @"12", @"color" : @"1"}];
    [self.tabView initWithButtonList: buttonList];
    [self.view addSubview:self.tabView];
    
    self.menuViewController = [[UNMenuViewController alloc] initWithNibName:@"UNMenuViewController" bundle:nil];
#if 0  // ジェスチャーテスト
    UIView *view1 = [[UIView alloc] initWithFrame:CGRectMake(100, 100, 100, 100)];
    view1.backgroundColor = [UIColor redColor];
    [self.view addSubview:view1];

    //ジェスチャーをUIViewオブジェクトに登録する
    UITapGestureRecognizer *recognizer = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(buttonTapped:)];
    [view1 addGestureRecognizer: recognizer];
    view1.userInteractionEnabled = YES;
#endif
    [self.view addSubview:self.menuViewController.view];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - Private method

-(void)buttonTapped:(id)sender
{
    NSLog(@"ok");
}

#pragma mark - Public method

- (IBAction)tableViewButtonDidTap:(id)sender {
    
    
}
@end

File: swift/UIViewTest/UIViewTest/ViewController/ViewController.swift
//
//  ViewController.swift
//  UIViewTest
//
//  Created by Shusuke Unno on 2016/08/27.
//  Copyright © 2016年 Shusuke Unno. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    var view1 : UIView?
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Viewを追加
        self.view1 = UIView(frame: CGRect(x:0, y:0, width:100, height:100))
        view1!.backgroundColor = UIColor.redColor()
        self.view.addSubview(view1!)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
